### PR TITLE
CI: manage stale PRs

### PR DESCRIPTION
**Pull request by adeebshihadeh**
I think we want to whitelist issues with a tag and blacklist PRs with a tag.

### CODE REVIEW

**Comment by jnewb1** [comment]
what do you think of these rules for stale PRs:

- 30 days inactive + 7 days stale = close pr
- only delete branches that are on the main repo
- exclude "needs testing" (for stuff that we just need someone in the community to test for us, like a new car port) and "work in progress" (stuff we expect will take a while and the PR is only open for encouraging others to work on it or that we might get around to eventually)
- comments or pushes don't reset the timer, which would encourage stuff to get merged faster

File: .github/workflows/stale.yaml [hunk 1]
```diff
@@ -15,10 +15,13 @@ jobs:
...
+          days-before-issue-stale: -1 # ignore issues for now
+          delete-branch: ${{ github.event.pull_request.head.repo.full_name != 'commaai/openpilot' }} # only delete branches on the main repo
           debug-only: true
           exempt-milestones: true
-          exempt-pr-labels: "Work In Progress"
+          only-issue-labels: "bug" # only close issues with "bug" label
+          exempt-pr-labels: "Work In Progress" # if wip, don't mark as stale
           days-before-stale: ${{ env.DAYS_BEFORE_STALE }}
           days-before-close: ${{ env.DAYS_BEFORE_CLOSE }}
+          ignore-update: true # comments or pushes won't reset the timer, encourage stuff to get merged faster
```
Comment (adeebshihadeh) [inline .github/workflows/stale.yaml:26]: let's start off more conservative and let this stuff reset it

File: .github/workflows/stale.yaml [hunk 2]
```diff
@@ -15,10 +15,13 @@ jobs:
...
         with:
           stale-pr-message: 'This PR has had no activity for ${{ env.DAYS_BEFORE_STALE }} days. It will be automatically closed in ${{ env.DAYS_BEFORE_CLOSE }} days.'
-          delete-branch: ${{ github.event.pull_request.head.repo.full_name != 'commaai/openpilot' }} # only delete branches on the main repo 
+          days-before-issue-stale: -1 # ignore issues for now
+          delete-branch: ${{ github.event.pull_request.head.repo.full_name != 'commaai/openpilot' }} # only delete branches on the main repo
           debug-only: true
           exempt-milestones: true
-          exempt-pr-labels: "Work In Progress"
+          only-issue-labels: "bug" # only close issues with "bug" label
+          exempt-pr-labels: "Work In Progress" # if wip, don't mark as stale
```
Comment (adeebshihadeh) [inline .github/workflows/stale.yaml]:
```suggestion
          exempt-pr-labels: "active-work-in-progress"
```

File: .github/workflows/stale.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  DAYS_BEFORE_STALE: 30
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v8
+        with:
+          stale-pr-message: 'This PR has had no activity for ${{ env.DAYS_BEFORE_STALE }} days. It will be automatically closed in ${{ env.DAYS_BEFORE_CLOSE }} days.'
+          close-pr-message: 'This PR has been automatically closed due to inactivity.'
```
Comment (adeebshihadeh) [inline .github/workflows/stale.yaml:18]:
```suggestion
          close-pr-message: 'This PR has been automatically closed due to inactivity. Feel free to re-open once activity resumes.'
```

**Comment by adeebshihadeh** [comment]
Before merging this, I'd like to look over the affected PRs. @sshane can you roughly start assigning them to people? if not sure about one, you can assign to me

File: .github/workflows/stale.yaml (hunk 2 above)
Comment (jnewb1) [reply .github/workflows/stale.yaml]: "active work in progress" fine? we don't have hyphens in any other label

File: .github/workflows/stale.yaml (hunk 3 above)
Comment (jnewb1) [reply .github/workflows/stale.yaml]: done

File: .github/workflows/stale.yaml (hunk 1 above)
Comment (jnewb1) [reply .github/workflows/stale.yaml]: done

**Comment by jnewb1** [comment]
our stale PR PR is about to go stale 😅

- [ ] #28743
- [ ] #28641
- [ ] #28344
- [ ] #27915
- [ ] #27586
- [ ] #27373

**Pull request merged**